### PR TITLE
feat: Add enable deletion protection and drop invalid header fields

### PR DESCRIPTION
**Pull request by OneBadSanta**
## Description
This PR allows for configuring the enable_deletion_protection and drop_invalid_header_fields for the ALB. Its variables default to false, which is the default ALB module.

## Motivation and Context
I needed to configure these options for compliance reasons.

## Breaking Changes
No

## How Has This Been Tested?
Tested on own infrastructure

### CODE REVIEW

**Comment by OneBadSanta** [comment]
@antonbabenko Are there any additional changes I need to make or does what is submitted look sufficient? Thx

File: examples/github-complete/variables.tf
Comment (bryantbiggs) [inline examples/github-complete/variables.tf:17]: you can remove these two variables from the example variables

File: examples/github-complete/main.tf
Comment (bryantbiggs) [inline examples/github-complete/main.tf:84]: lets remove this line from the example

File: examples/github-complete/main.tf
Comment (bryantbiggs) [inline examples/github-complete/main.tf:85]: lets set this line to `true`:
```suggestion
  alb_drop_invalid_header_fields  = true
```

File: examples/github-complete/variables.tf
Comment (bryantbiggs) [inline examples/github-complete/variables.tf:20]: lets set these to `null` for defaults to rely on whatever is setup upstream or by the API
```suggestion
  default     = null
```

File: examples/github-complete/variables.tf
Comment (bryantbiggs) [inline examples/github-complete/variables.tf:26]:
```suggestion
  default     = null
```

File: variables.tf
Comment (bryantbiggs) [inline variables.tf:126]: lets set these to default to `null` so that they default to what is provided upstream by the provider or by the AWS API:
```suggestion
  default     = null
```

File: variables.tf
Comment (bryantbiggs) [inline variables.tf:132]:
```suggestion
  default     = null
```

**Comment by OneBadSanta** [comment]
@bryantbiggs Changes sound good to me. Please see updates

**Review by bryantbiggs — APPROVED** [review]
@antonbabenko 👍🏼

**Comment by OneBadSanta** [comment]
Thanks @bryantbiggs

File: variables.tf
Comment (antonbabenko) [inline variables.tf:126]: These two variables should have default `false` since this is what the ALB module has as defaults - https://github.com/terraform-aws-modules/terraform-aws-alb/blob/master/variables.tf#L7-L17

Btw, description is correctly saying `false` :)

If we change this to `null` it will require an update for everyone using this module.

Comment (bryantbiggs) [reply variables.tf:126]: the ALB module makes assumptions by setting `false` as default, but the underlying provider/API ultimately defaults to `false` as well. I was trying to avoid making any assumptions with defaults if they are not required:
image
image

Comment (OneBadSanta) [reply variables.tf:126]: @antonbabenko  Personally, I can go either way. However, what @bryantbiggs suggested made sense to me in that setting the values to null is relying on AWS to provide these reasonable or possible preferable defaults. Setting them to default as false in this modules opinionates it further (which may or may not be preferred). 

I'm not entirely clear on how setting this to null will require everyone to update, but that does not sound like a good time.

Like I mentioned, I can go either way. I have only been using this module a couple weeks so I defer to the maintainers judgement. Please tell me what changes to make and I will make them. Thanks

Comment (antonbabenko) [reply variables.tf:126]: I misread all this. You are right, it should be `null`, my bad 👍  Merging right away!

**Comment by antonbabenko** [comment]
Thank you guys for this PR!

[v2.38.0](https://github.com/terraform-aws-modules/terraform-aws-atlantis/releases/v2.38.0) has been just released.

**Pull request merged**
